Breakout functions (#47)
datacarpentry/astronomy-python: _extras/guide.md

* This lesson takes a slight detour to introduce a few features of Pandas DataFrames. To keep this connected to the story, you can talk about how data exploration is an important part of prototyping your query and making sure you are getting the results you expect in the format you expect them in.
* Starting with selecting the centerline, we do a series of filters on different data frames. Take a minute before you teach this section to make sure you understand what each one represents. We use results_df to build centerline_df. We use centerline_df to determine proper motion limits. We use the proper motion limits determined from centerline_df to select GD-1 stars from results_df. This is selected_df.

* It is also worth noting that we are selecting stars close to the centerline to identify the proper motion cut, so its ok if we exclude some GD-1 stars here. In this case we prefer a more pure sample to a more complete sample. Once we have the proper motion limits from our pure sample (centerline_df), we'll include the full spatial region (results_df) and get all of the GD-1 stars (selected_df).

> * Learners may be concerned that we're writing a function for later that doesn't fill the full canvas (when we set `axis('equal')`). You can reassure them that in episode 7 we'll take care of this by learning how to set the figure size, the subplot size, and we'll be using a larger spatial area (that we define in episode 4).

Reviewer: ```suggestion
* Learners may be concerned that we are writing a function for later that does not fill the full canvas (when we set `axis('equal')`). You can reassure them that in episode 7 we will take care of this by learning how to set the figure size, the subplot size, and we will be using a larger spatial area (that we define in episode 4).
```

* Recall that the first time we use `DataFrame.to_hdf`, we use the `w` argument to indicate that we want to create a new, empty HDF Store.  For all subsequent uses, we should *not* use the `w` argument, so that we add new Datasets to the existing Store, rather than starting over.

> * when writing the point_series object to an HDF5 file, learners may see the warning message like the following:

Reviewer: ```suggestion
* when writing the `point_series` object to an HDF5 file, learners may see the warning message like the following:
```

* The Pan-STARRS join exercise is likely to feel scary to a lot of students - they have only seen one example. Emphasize the building blocks of a join e.g. FROM table1 JOIN table2 ON table1.common_column=table2.common_column. It is also worth pointing out that common_columns will not have the same names in this case

> * The `head` function is a python version of the Unix `head`. The Python version (which can be found in the episode_functions.py file) is used because it is platform independent so a single syntax can be used by all learners. You don't need to go through the function, but if someone asks, feel free to open the episode functions.py file and show them. 

Reviewer: ```suggestion
* The `head` function is a python version of the Unix `head` command. The Python version (which can be found in the episode_functions.py file) is used because it is platform independent so a single syntax can be used by all learners. You don't need to go through the function, but if someone asks, feel free to open the episode functions.py file and show them. 
```
